components/experimental/src/dimension/units/format.rs

```diff
@@ -71,6 +71,23 @@ fn test_basic() {
     formatted_unit.write_to(&mut sink).unwrap();
     assert_eq!(sink.as_str(), "12,345.67 m");
 
+    let locale = locale!("en-US").into();
+    let century = "century";
+    let fmt = UnitsFormatter::try_new(
+        &locale,
+        century,
+        UnitsFormatterOptions {
+            width: Width::Long,
+            ..Default::default()
+        },
+    )
+    .unwrap();
+    let value = "12345.67".parse().unwrap();
+    let formatted_unit = fmt.format_fixed_decimal(&value, meter);
+    let mut sink = String::new();
+    formatted_unit.write_to(&mut sink).unwrap();
+    assert_eq!(sink.as_str(), "12,345.67 centuries");
```

Comment:
```suggestion
    assert_writeable_eq!(formatted_unit, "12,345.67 centuries");
```

Reply:
I have done it here: https://github.com/unicode-org/icu4x/pull/5229